test/lint/pending-lint-includes.sh

```diff
@@ -71,7 +71,6 @@ EXPECTED_BOOST_INCLUDES=(
     boost/test/unit_test.hpp
     boost/thread.hpp
     boost/thread/condition_variable.hpp
-    boost/thread/mutex.hpp
     boost/thread/thread.hpp
```

Comment:
Are other `boost/thread/xxx` okay as they are?

Reply:
these are all removed in #270.